src/hdtXSDValidator.cpp

```
@@ -0,0 +1,826 @@
+
+		if (tc.base == TypeConstraint::Base::Boolean) {
+			if (val != "true" && val != "false" && val != "1" && val != "0")
+				ctx.addViolation(row, col, where + " has invalid boolean value '" + val + "'");
+			return;
+		}
+
+		if (tc.base == TypeConstraint::Base::Float || tc.base == TypeConstraint::Base::Integer) {
+			double dval = 0.0;
+			const auto [ptr, ec] = std::from_chars(val.data(), val.data() + val.size(), dval);
+			if (ec != std::errc{} || ptr != val.data() + val.size()) {
+				const std::string typeName = (tc.base == TypeConstraint::Base::Float) ? "float" : "integer";
```

Clang (14.0.6)

[warning] variable 'typeName' is not initialized

(cppcoreguidelines-init-variables)